application/client.core/src/app/environment/components/dialogs/multiplefiles/component.ts

```diff
@@ -1,26 +1,86 @@
-import { Component, ChangeDetectorRef, Input, AfterContentInit } from '@angular/core';
-
-interface IFileInfo {
-    name: string;
-    modified: number;
-    modifiedStr: string;
-    size: number;
-}
+import { Component, ChangeDetectorRef, Input, AfterViewInit, ViewContainerRef, OnDestroy } from '@angular/core';
+import ServiceElectronIpc, { IPCMessages } from '../../../services/service.electron.ipc';
+import { FilesList } from '../../../controller/controller.file.storage';
+import { Subscription } from 'rxjs';
+import { ControllerComponentsDragDropFiles } from '../../../controller/components/controller.components.dragdrop.files';
+import * as Toolkit from 'chipmunk.client.toolkit';
 
 @Component({
     selector: 'app-views-dialogs-multiplefilescation-map',
     templateUrl: './template.html',
     styleUrls: ['./styles.less']
 })
 
-export class DialogsMultipleFilesActionComponent implements AfterContentInit {
+export class DialogsMultipleFilesActionComponent implements AfterViewInit, OnDestroy {
+
+    @Input() files: IPCMessages.IFile[] = [];
+    @Input() fileList: FilesList;
+
+    private _dragdrop: ControllerComponentsDragDropFiles | undefined;
+    private _subscriptions: { [key: string]: Subscription | undefined } = { };
+    private _logger: Toolkit.Logger = new Toolkit.Logger('MultipleFilesDialog');
+    private _destroyed = false;
+
+    constructor(private _cdRef: ChangeDetectorRef, private _vcRef: ViewContainerRef, ) {
+    }
+
+    ngAfterViewInit() {
+        this._dragdrop = new ControllerComponentsDragDropFiles(this._vcRef.element.nativeElement);
+        this._subscriptions.onFiles = this._dragdrop.getObservable().onFiles.subscribe(this._onFilesDropped.bind(this));
+    }
 
-    @Input() files: File[] = [];
+    ngOnDestroy() {
+        this._destroyed = true;
+        Object.keys(this._subscriptions).forEach((key: string) => {
+            this._subscriptions[key].unsubscribe();
+        });
+    }
+
+    private _forceUpdate() {
+        if (this._destroyed) {
+            return;
+        }
+        this._cdRef.detectChanges();
+    }
+
+    private _getUniqueFiles(files: IPCMessages.IFile[]): Promise<IPCMessages.IFile[]> {
+        return new Promise((resolve) => {
+            const paths: string[] = [];
+            let unique_files: IPCMessages.IFile[] = [];
+
+            this.files.forEach((file: IPCMessages.IFile) => {
+                paths.push(file.path);
+            });
+
+            unique_files = files.filter((ifile: IPCMessages.IFile) => paths.indexOf(ifile.path) === -1 );
+            resolve(unique_files);
+        });
+    }
```

Comment:
Yeah, the code works and I get the idea, but I think you misunderstood the function because there are two different arrays that are compared (this.files and files).
Your suggestion filters out duplicate files out of the list, which should never be that case since every file has a unique path (as far as I know). The function above compares to the ```this.files``` list so we don't have the same files in the dialog in case the user accidentally throws the same files into the dialog again.

I think this we should modify it to this:
```
private _getUniqueFiles(files: IPCMessages.IFile[]): IPCMessages.IFile[] {
    return files.filter((file: IPCMessages.IFile) => {
        return this.files.findIndex((f: IPCMessages.IFile) => f.path === file.path) === -1;
    });
}
```